platform/xbox/include/pdclib/_PDCLIB_config.h

```diff
@@ -137,8 +137,8 @@ struct _PDCLIB_lldiv_t
 /* -------------------------------------------------------------------------- */
 
 /* The result type of substracting two pointers */
-#define _PDCLIB_ptrdiff long
-#define _PDCLIB_PTRDIFF LONG
+#define _PDCLIB_ptrdiff int
+#define _PDCLIB_PTRDIFF INT
```

Comment:
Where did the original code originate? Is this wrong in the upstream sample?

Reply:
I originally copied that part from the upstream sample, but that targets another platform of course (normal Linux). I did not check whether `long` is correct for the sample.

Reply:
On Discord, I asked:

> Where does ptrdiff_t come from, and why isn't it the same as for Linux?
> That linux code doesn't seem to be arch dependent?
> And why does the C++ lib not have an issue in linux then?

The response by @thrimbor was:

> The C standard only says that ptrdiff_t "is the signed integer type of the result of subtracting two pointers", and size_t "is the unsigned integer type of the result of the sizeof opereator".
> The actual types used for these are platform dependent, Linux uses long / unsigned long, Windows uses int / unsigned int (just checked VS headers).
> libc++ doesn't seem to have a problem unless those two are mismatched.
> As to where exactly it gets the types from - no idea, code is large and complex, so didn't bother once I found the issue.
>
> ftr, code breaks here https://github.com/llvm-mirror/libcxx/blob/master/include/streambuf#L413 and here https://github.com/llvm-mirror/libcxx/blob/master/include/streambuf#L471